editor/tree_search.cpp

```diff
@@ -0,0 +1,492 @@
+		ERR_FAIL_NULL(stylebox);
+
+		// extract separation
+		float h_sep = p_tree_item->get_tree()->get_theme_constant("h_separation");
+
+		// compose draw rect
+		const Vector2 PADDING = Vector2(4., 2.);
+		Rect2 draw_rect = p_rect;
+
+		Vector2 rect_offset = Vector2(substring_before_size.x, 0);
+		rect_offset.x += p_tree_item->get_icon_max_width(0);
+		rect_offset.x += (h_sep + 4. * EDSCALE); // TODO: Find better way to determine texts x-offset
+		rect_offset.y = (p_rect.size.y - substring_match_size.y) / 2; // center box vertically
+
+		draw_rect.position += rect_offset - PADDING / 2;
+		draw_rect.size = substring_match_size + PADDING;
+
+		// draw
+		stylebox->draw(p_tree_item->get_tree()->get_canvas_item(), draw_rect);
+	}
+
+	// second part: draw number
+	int num_mat = number_matches.has(p_tree_item) ? number_matches.get(p_tree_item) : 0;
+	if (num_mat > 0) {
+		float h_sep = p_tree_item->get_tree()->get_theme_constant("h_separation");
+		Ref<Font> font = tree_reference->get_theme_font("font");
+		float font_size = tree_reference->get_theme_font_size("font") * 0.75;
+
+		String num_string = String::num_int64(num_mat);
+		Vector2 string_size = font->get_string_size(num_string, HORIZONTAL_ALIGNMENT_CENTER, -1, font_size);
+		Vector2 text_pos = p_rect.position;
+
+		text_pos.x += p_rect.size.x - string_size.x - h_sep;
+		text_pos.y += font->get_descent(font_size) + p_rect.size.y / 2.; // center vertically
+
+		font->draw_string(tree_reference->get_canvas_item(), text_pos, num_string, HORIZONTAL_ALIGNMENT_CENTER, -1, font_size);
+	}
+}
+
+void TreeSearch::_update_matching_entries(const String &p_search_mask) {
+	Vector<TreeItem *> accum;
+	matching_entries = _find_matching_entries(tree_reference->get_root(), p_search_mask, accum);
+}
+
+/* this linearizes the tree into [ordered_tree_items] like so:
+ - i1
+	- i2
+	- i3
+ - i4 ---> [i1,i2,i3,i4]
+*/
+void TreeSearch::_update_ordered_tree_items(TreeItem *p_tree_item) {
+	if (!p_tree_item)
+		return;
+	if (p_tree_item == p_tree_item->get_tree()->get_root()) {
+		ordered_tree_items.clear();
+	}
+	// Add the current item to the list
+	ordered_tree_items.push_back(p_tree_item);
+
+	// Recursively collect items from the first child
+	TreeItem *child = p_tree_item->get_first_child();
+	while (child) {
+		_update_ordered_tree_items(child);
+		child = child->get_next();
+	}
+}
+
+void TreeSearch::_update_number_matches() {
+	number_matches.clear();
+	for (int i = 0; i < matching_entries.size(); i++) {
+		TreeItem *item = matching_entries[i];
+		while (item) {
+			int old_num_value = number_matches.has(item) ? number_matches.get(item) : 0;
+			number_matches[item] = old_num_value + 1;
+			item = item->get_parent();
+		}
+	}
+}
+
+String TreeSearch::_get_search_mask() {
+	ERR_FAIL_COND_V(!search_panel, "");
+	return search_panel->get_text();
+}
+
+Vector<TreeItem *> TreeSearch::_find_matching_entries(TreeItem *p_tree_item, const String &p_search_mask, Vector<TreeItem *> &p_accum) {
```

Comment:
Argument name differs in the header. If it's a return argument, it should be called `r_accum` or `r_buffer` - this is how it's done in other places. There seems to be no need to explicitly return it, either.

Reply:
Inconsistency resolved. Is now void.